.github/workflows/build-and-test.yml

```diff
@@ -15,75 +15,59 @@ on:
   pull_request:
   workflow_dispatch:
 
+concurrency:
+  group: Build-and-Test-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
-  buildandtest_ios:
-    name: Build and Test Swift Package iOS
-    uses: StanfordSpezi/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
-    with:
-      runsonlabels: '["macOS", "self-hosted"]'
-      scheme: Spezi-Package
-      resultBundle: Spezi-Package-iOS.xcresult
-      artifactname: Spezi-Package-iOS.xcresult
-  buildandtest_watchos:
-    name: Build and Test Swift Package watchOS
-    uses: StanfordSpezi/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
-    with:
-      runsonlabels: '["macOS", "self-hosted"]'
-      scheme: Spezi-Package
-      destination: 'platform=watchOS Simulator,name=Apple Watch Series 10 (46mm)'
-      resultBundle: Spezi-Package-watchOS.xcresult
-      artifactname: Spezi-Package-watchOS.xcresult
-  buildandtest_visionos:
-    name: Build and Test Swift Package visionOS
-    uses: StanfordSpezi/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
-    with:
-      runsonlabels: '["macOS", "self-hosted"]'
-      scheme: Spezi-Package
-      destination: 'platform=visionOS Simulator,name=Apple Vision Pro'
-      resultBundle: Spezi-Package-visionOS.xcresult
-      artifactname: Spezi-Package-visionOS.xcresult
-  buildandtest_tvos:
-    name: Build and Test Swift Package tvOS
-    uses: StanfordSpezi/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
-    with:
-      runsonlabels: '["macOS", "self-hosted"]'
-      scheme: Spezi-Package
-      destination: 'platform=tvOS Simulator,name=Apple TV 4K (3rd generation)'
-      resultBundle: Spezi-Package-tvOS.xcresult
-      artifactname: Spezi-Package-tvOS.xcresult
-  buildandtest_macos:
-    name: Build and Test Swift Package macOS
-    uses: StanfordSpezi/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
+  package_tests:
+    name: Build and Test Swift Package ${{ matrix.platform.name }} (${{ matrix.config }})
+    uses: StanfordBDHG/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
+    strategy:
+      matrix:
+        config: [Debug, Release]
```

Comment:
you'll likely run into the same release testing but i ran into in #130 (that's the reason why that PR still hasn't been merged)

IMO we should, for the time being, remove the Release config from the matrix inputs and then fix this in a follow-up PR